modules/fedimint-ln-client/src/incoming.rs

```diff
@@ -177,41 +172,12 @@ impl FundingOfferState {
 
     async fn await_funding_success(
         global_context: DynGlobalClientContext,
-        out_point: OutPoint,
-        context: LightningClientContext,
+        txid: TransactionId,
     ) -> Result<(), IncomingSmError> {
-        debug!("Awaiting funding success for outpoint: {out_point:?}");
-        for retry in 0.. {
-            let sleep = (retry * 15).min(90);
-            match global_context
-                .api()
-                .await_output_outcome::<LightningOutputOutcome>(
-                    out_point,
-                    Duration::from_secs(90),
-                    &context.ln_decoder,
-                )
-                .await
-            {
-                Ok(_) => {
-                    debug!("Funding success for outpoint: {out_point:?}");
-                    return Ok(());
-                }
-                Err(e) if e.is_rejected() => {
-                    warn!("Funding failed for outpoint: {out_point:?}: {e:?}");
-                    return Err(IncomingSmError::FailedToFundContract {
-                        error_message: e.to_string(),
-                    });
-                }
-                Err(e) => {
-                    e.report_if_important();
-                    debug!(error = %e, "Awaiting output outcome failed, retrying in {sleep}s",);
-                }
-            }
-            // give some time for other things to run
-            fedimint_core::runtime::sleep(Duration::from_secs(sleep)).await;
-        }
-
-        unreachable!("there is too many u64s to ever get here")
+        global_context
+            .await_tx_accepted(txid)
+            .await
+            .map_err(|error_message| IncomingSmError::FailedToFundContract { error_message })
```

Comment:
Seems we used `await_output_outcome` while it could have been the much more lightweight `await_tx_accepted` all along, worth considering in the future. cc @m1sterc001guy @tvolk131